File: Leetcode/2874 - Maximum Value of an Ordered Triplet II/solution.js
var maximumTripletValue = function(nums) {
    let numsLength = nums.length;
    let maxTripletScore = 0;
    let prefixMax = new Array(numsLength).fill(0);
    prefixMax[0] = nums[0];
    let suffixMax = new Array(numsLength).fill(0);
    suffixMax[numsLength-1] = nums[numsLength-1];
    for (let x = 1; x < numsLength; x++) {
        prefixMax[x] = Math.max(nums[x], prefixMax[x-1]);
    }
    for (let y = numsLength-2; y >= 0; y--) {
        suffixMax[y] = Math.max(nums[y], suffixMax[y + 1]);
    }
    for (let j = 1; j < nums.length-1; j++) {
        let tripletScore = (prefixMax[j-1] - nums[j]) * suffixMax[j+1];
        maxTripletScore = Math.max(maxTripletScore, tripletScore);
    }
    return maxTripletScore;
};

File: Leetcode/2874 - Maximum Value of an Ordered Triplet II/solution.py
def maximumTripletValue(nums):
    nums_length = len(nums)
    max_triplet_score = 0
    prefix_max = [0] * nums_length
    prefix_max[0] = nums[0]
    suffix_max = [0] * nums_length
    suffix_max[nums_length - 1] = nums[nums_length - 1]

    for x in range(1, nums_length):
        prefix_max[x] = max(nums[x], prefix_max[x - 1])
    for y in range(nums_length - 2, -1, -1):
        suffix_max[y] = max(nums[y], suffix_max[y + 1])
    for j in range(1, nums_length - 1):
        triplet_score = (prefix_max[j - 1] - nums[j]) * suffix_max[j + 1]
        max_triplet_score = max(max_triplet_score, triplet_score)

    return max_triplet_score